graphql.go

```diff
@@ -122,6 +142,29 @@ func (c *Client) buildQueryAndOptions(op operationType, v any, variables map[str
 
 // Request the common method that send graphql request
 func (c *Client) request(ctx context.Context, query string, variables map[string]any, options *constructOptionsOutput) ([]byte, []byte, *http.Response, io.Reader, Errors) {
+	attempts := 0
+	for {
+		attempts++
```

Comment:
Move this line to the end of the loop. The first request isn't counted as a retry